qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,705 @@
+                            valid_c_op_products,
+                            np.eye(Hdim, Hdim),
+                            np.eye(Hdim, Hdim),
+                            np.eye(Hdim, Hdim),
+                            np.eye(Hdim, Hdim))
+            try:
+                total_R_tensor[key] += np.reshape(flime_FirstTerm - (1 / 2)
+                                                  * (flime_SecondTerm +
+                                                     flime_ThirdTerm),
+                                                  (Hdim**2, Hdim**2))
+            except KeyError:
+                total_R_tensor[key] = np.reshape(flime_FirstTerm - (1 / 2)
+                                                 * (flime_SecondTerm +
+                                                    flime_ThirdTerm),
+                                                 (Hdim**2, Hdim**2))
+    return total_R_tensor
+
+
+def flimesolve(
+        H,
+        rho0,
+        taulist,
+        T,
+        Nt=None,
+        c_ops_and_rates=[],
+        e_ops=[],
+        args=None,
+        time_sense=0,
+        quicksolve=False,
+        options=None):
+    """
+    Parameters
+    ----------
+
+    H : :class:`Qobj`,:class:`QobjEvo`,:class:`QobjEvo` compatible format.
+        Periodic system Hamiltonian as :class:`QobjEvo`. List of
+        [:class:`Qobj`,:class:`Coefficient`] or callable that
+        can be made into :class:`QobjEvo` are also accepted.
+
+    rho0 / psi0 : :class:`qutip.Qobj`
+        Initial density matrix or state vector (ket).
+
+    Taulist:*list* / *array*
+        List of times for :math:`t`.
+
+    T : float
+        The period of the time-dependence of the hamiltonian.
+
+    c_ops_and_rates : list of :class:`qutip.Qobj`.
+        List of lists of [collapse operator,collapse operator rate] pairs
+
+    e_ops : list of :class:`qutip.Qobj` / callback function
+        List of operators for which to evaluate expectation values.
+        The states are reverted to the lab basis before applying the
+
+    args : *dictionary*
+        Dictionary of parameters for time-dependent Hamiltonian
+
+    time_sense : float
+        Experimental. Value of the secular approximation (in terms of system
+        frequency 2*np.pi/T) to use when constructing the rate matrix R(t).
+        Default value of zero uses the fully time-independent/most strict
+        secular approximation.
+
+    quicksolve: Boolean
+        True to use the quicksolve method,which utilizes the most strict
+        secular approximation to quickly solve the system dynamics with a
+        matrix-exponential method instead of QuTiP internal IVP solvers.
+        Automatically updates based on time_sense,with the default time_sense
+        value 0 using quicksolve,and any other time-sense value using the
+        IVP solvers. Can be overridden if desired (e.g. for
+                                                   debuggin/troubleshooting).
```

Comment:
I think I may end up writing out time_solve as an input argument altogether. It can be overwritten by making quicksolve = True, which will use the quicksolve method even if a time dependence is desired. However, I feel like this may add confusion and ultimately doesn't provide any extra functionality, such that I don't think it'll be worthwhile to keep this as it is.